core/arch/arm/kernel/boot.c

```diff
@@ -1067,6 +1067,108 @@ static int mark_tzdram_as_reserved(struct dt_descriptor *dt)
 				   CFG_TZDRAM_SIZE);
 }
 
+#ifdef CFG_CORE_RESERVE_DRIVERS
+static char *strtok(char *str, const char *delims)
+{
+	static char *pos;
+	char *start = NULL;
+
+	if (str)
+		pos = str;
+
+	if (pos) {
+		while (*pos && strchr(delims, *pos))
+			pos++;
+
+		if (*pos) {
+			start = pos;
+			while (*pos && !strchr(delims, *pos))
+				pos++;
+
+			if (*pos)
+				*pos++ = '\0';
+		}
+	}
+
+	return start;
+}
+
+static int add_descriptor_node(struct dt_descriptor *dt, const char *driver)
+{
+	unsigned int i = 0;
+	unsigned int j = 0;
+	char root_name[50] = { '\0' };
+	char buffer[200] = { '\0' };
+	char *item[10] = { NULL };
+	char *root = NULL;
+	int offs = 0;
+
+	/* strok will modify the buffer */
+	strncpy(buffer, driver, sizeof(buffer));
+
+	root = strtok(buffer, "/");
+	if (!root)
+		return -1;
+
+	for (i = 0; i < ARRAY_SIZE(item); i++) {
+		item[i] = strtok(NULL, "/");
+		if (!item[i])
+			break;
+	}
+
+	if (i >= ARRAY_SIZE(item))
+		return -1;
+
+	if (fdt_path_offset(dt->blob, driver) >= 0)
+		return 0;
```

Comment:
If the node exists, don't you want to change it to `status="disabled"; secure-status='okay";`?
If the node does not exist, why adding it with a disabled status? (I assume these changes target the FDT that will be used by non-secure world/bootloader).

Reply:
@etienne-lms right, so 
* if the node exists, yes we do not add the node and just change the status to disabled and the secure-status to okay. That is what the code does AFAICS?
* if the node does not exist, we do create the node anyways just so anyone inspecting sysfs is aware that the secure world is using the device.